### DISCUSSION TITLE
Multiple editions/versions of the same document

### DISCUSSION THREAD

**Discussion by fengkaijia** [jonaswinkler/paperless-ng#773] [Feature Requests]
Great thanks to @jonaswinkler and @danielquinn for this amazing project.

I would like to see Paperless handling multiple editions or versions of the same document. I could think of two scenarios:

1. For something very important and frequently used, such as some kinds of verification documents or a diploma. A user probably tends to scan a very high DPI version so record keeping, and then another low DPI version to use in daily verification.
2. For something a user scanned sometimes ago with a low-quality scanner, and then years later, scanned again with a more advanced scanner. The user might want to keep both copies, such as the case when the ink might have faded during the latest scan, so it's reasonable to keep the old one (although having a low DPI) as the text might be more clear.

In both cases, two scanned copies are for the same document, better keep them under the same roof than as separated entries.

**Comment by jonaswinkler** [comment]
Hello!

I won't add versioning in the near future. Paperless was never designed to support this, and many parts of paperless work on the assumption that one document in paperless has exactly one file on disk. Changing that would require major changes to many components (and associated test cases). *Maybe* later.

I'd much rather focus on improving what's actually there before adding too many bells and whistles. I also have to consider that each added feature increases the overall size of the code base, which in turn means more required time for maintenance, and I've got a full time job as well :)

**Comment by krikkert** [comment]
Could also be implemented via document relationships (#564 ) or custom metadata (#274 ). "Superseded by (doc number)" and "Supersedes (doc number)".
